File: Question1.cs
using System;
namespace Algorithm_Practice{
    class Question1 {
        public Question1() {
            
        }

        public void Action (int n) {
            Console.WriteLine("{0}! = {1}", n, Fact(n));
        }

        private int Fact (int n) {
            if (n == 0) {
                return 1;
            } else {
                return n * Fact(n - 1);
            }
        }
    }
}

File: Question1.java
class Question1 {
	// 正の整数nの階乗を求める(再帰処理)
	public static void main(String[] args) {
		// TODO 自動生成されたメソッド・スタブ
		final int NUM = 5;
		System.out.println(NUM + "! = " + fact(NUM));
	}

	static int fact(int num) {
		if (num == 1) {
			return num;
		} else {
			return num * fact(num - 1);
		}
	}

}
